tests/integration/telemetry/stats/prometheus/customizemetrics/customize_metrics_test.go

```diff
@@ -61,6 +63,11 @@ func TestCustomizeMetrics(t *testing.T) {
 		Features("observability.telemetry.request-classification").
 		Features("extensibility.wasm.remote-load").
 		Run(func(ctx framework.TestContext) {
+			ctx.Cleanup(func() {
+				if ctx.Failed() {
+					util.PromDump(ctx.Clusters().Default(), promInst, "istio_requests_total")
```

Comment:
Before we do PromDump (slow) on every for an iteration. Instead just dump it if the actual test fails

Reply:
iirc, this was a legacy from 0.1 and the notion that having golden logs to inspect for differences in case of failures was somehow worthwhile.